src/main/drivers/serial_uart_stm32h7xx.c

```diff
@@ -0,0 +1,567 @@
+#define UART3_RX_DMA_STREAM NULL
+#endif
+#ifndef UART4_TX_DMA_STREAM
+#define UART4_TX_DMA_STREAM NULL
+#endif
+#ifndef UART4_RX_DMA_STREAM
+#define UART4_RX_DMA_STREAM NULL
+#endif
+#ifndef UART5_TX_DMA_STREAM
+#define UART5_TX_DMA_STREAM NULL
+#endif
+#ifndef UART5_RX_DMA_STREAM
+#define UART5_RX_DMA_STREAM NULL
+#endif
+#ifndef UART6_TX_DMA_STREAM
+#define UART6_TX_DMA_STREAM NULL
+#endif
+#ifndef UART6_RX_DMA_STREAM
+#define UART6_RX_DMA_STREAM NULL
+#endif
+#ifndef UART7_TX_DMA_STREAM
+#define UART7_TX_DMA_STREAM NULL
+#endif
+#ifndef UART7_RX_DMA_STREAM
+#define UART7_RX_DMA_STREAM NULL
+#endif
+#ifndef UART8_TX_DMA_STREAM
+#define UART8_TX_DMA_STREAM NULL
+#endif
+#ifndef UART8_RX_DMA_STREAM
+#define UART8_RX_DMA_STREAM NULL
+#endif
+
+#ifdef USE_UART1
+DMA_RAM volatile uint8_t uart1RxBuffer[UART_RX_BUFFER_SIZE];
+DMA_RAM volatile uint8_t uart1TxBuffer[UART_TX_BUFFER_SIZE];
+#endif
+
+#ifdef USE_UART2
+DMA_RAM volatile uint8_t uart2RxBuffer[UART_RX_BUFFER_SIZE];
+DMA_RAM volatile uint8_t uart2TxBuffer[UART_TX_BUFFER_SIZE];
+#endif
+
+#ifdef USE_UART3
+DMA_RAM volatile uint8_t uart3RxBuffer[UART_RX_BUFFER_SIZE];
+DMA_RAM volatile uint8_t uart3TxBuffer[UART_TX_BUFFER_SIZE];
+#endif
+
+#ifdef USE_UART4
+DMA_RAM volatile uint8_t uart4RxBuffer[UART_RX_BUFFER_SIZE];
+DMA_RAM volatile uint8_t uart4TxBuffer[UART_TX_BUFFER_SIZE];
+#endif
+
+#ifdef USE_UART5
+DMA_RAM volatile uint8_t uart5RxBuffer[UART_RX_BUFFER_SIZE];
+DMA_RAM volatile uint8_t uart5TxBuffer[UART_TX_BUFFER_SIZE];
+#endif
+
+#ifdef USE_UART6
+DMA_RAM volatile uint8_t uart6RxBuffer[UART_RX_BUFFER_SIZE];
+DMA_RAM volatile uint8_t uart6TxBuffer[UART_TX_BUFFER_SIZE];
+#endif
+
+#ifdef USE_UART7
+DMA_RAM volatile uint8_t uart7RxBuffer[UART_RX_BUFFER_SIZE];
+DMA_RAM volatile uint8_t uart7TxBuffer[UART_TX_BUFFER_SIZE];
+#endif
+
+#ifdef USE_UART8
+DMA_RAM volatile uint8_t uart8RxBuffer[UART_RX_BUFFER_SIZE];
+DMA_RAM volatile uint8_t uart8TxBuffer[UART_TX_BUFFER_SIZE];
+#endif
+
+const uartHardware_t uartHardware[UARTDEV_COUNT] = {
+#ifdef USE_UART1
+    {
+        .device = UARTDEV_1,
+        .reg = USART1,
+#ifdef USE_DMA
+        .rxDMARequest = DMA_REQUEST_USART1_RX,
+        .rxDMAStream = UART1_RX_DMA_STREAM,
+        .txDMARequest = DMA_REQUEST_USART1_TX,
+        .txDMAStream = UART1_TX_DMA_STREAM,
+#endif
+        .rxPins = {
+            { DEFIO_TAG_E(PA10), GPIO_AF7_USART1 },
+            { DEFIO_TAG_E(PB7),  GPIO_AF4_USART1 },
+            { DEFIO_TAG_E(PB15), GPIO_AF4_USART1 },
+        },
+        .txPins = {
+            { DEFIO_TAG_E(PA9),  GPIO_AF7_USART1 },
+            { DEFIO_TAG_E(PB6),  GPIO_AF4_USART1 },
+            { DEFIO_TAG_E(PB14), GPIO_AF4_USART1 },
+        },
+        .rcc_apb2 = RCC_APB2(USART1),
+        .rxIrq = USART1_IRQn,
+        .txPriority = NVIC_PRIO_SERIALUART1_TXDMA,
+        .rxPriority = NVIC_PRIO_SERIALUART1,
+        .txBuffer = uart1TxBuffer,
+        .rxBuffer = uart1RxBuffer,
+        .txBufferSize = ARRAYLEN(uart1TxBuffer),
```

Comment:
(BTW: `sizeof` is prefered here - number of bytes is requested, not items)

Reply:
👍 ... but may go away in the future...